File: hw/gavilanez/u3/HW29Strategy/Strategy/src/main/java/ec/edu/espe/strategy/model/SortingContext.java

package ec.edu.espe.strategy.model;

import ec.edu.espe.strategy.controller.SortingStrategy;
import ec.edu.espe.strategy.model.*;
/**
 *
 * @author Kenny Gavilanez, Dev Dynasty, DCCO-ESPE
 */
public class SortingContext {
    private SortingStrategy ss;
    
    public void sort(int data[]){
        int size = data.length;
        ss = setSortStrategy(size);
        ss.sort(data);
    }
    
    public SortingStrategy setSortStrategy(int size){
        if(size > 0 && size < 30){
            System.out.println("\nSorting with the bubble algorithm");
            return new BubbleSort();
            
        } else if(size >= 30 && size < 100){
            System.out.println("\nSorting with the insetion algorithm");
            return new InsertionSort();
            
        }else if(size >= 100){
            System.out.println("\nSorting with the quick algorithm");
            return new QuickSort();
            
        }
        return null;
    }
}

File: exams/Vaca/U3/ExamUnit3/controller/SortingContext.js
const BubbleSort = require('../model/BubbleSort');
const InsertionSort = require('../model/InsertionSort');
const QuickSort = require('../model/QuickSort');
const MongoDBController = require('./MongoDBController');

class SortingContext {
    constructor() {
        this.strategy = null;
    }

    setSortStrategy(array) {
        if (array.length >= 2 && array.length <= 5) {
            this.strategy = new BubbleSort();
        } else if (array.length > 5 && array.length <= 10) {
            this.strategy = new InsertionSort();
        } else {
            this.strategy = new QuickSort();
        }
    }

    sort(array) {
        this.setSortStrategy(array);
        const sortedArray = this.strategy.sort(array);
        const sortType = this.strategy.constructor.name;
        
        MongoDBController.saveSortedArray(sortType, sortedArray);

        return { sortedArray, sortType };
    }
}

module.exports = SortingContext;
